src/function/list/operations/include/list_extract_operation.h

```diff
@@ -36,8 +36,12 @@ struct ListExtract {
     }
 
     static inline void operation(ku_string_t& str, int64_t& idx, ku_string_t& result) {
-        auto pos = idx > 0 ? min(idx, (int64_t)str.len) : max(str.len + idx, (int64_t)0) + 1;
-        result.set((char*)(str.getData() + pos - 1), 1 /* length */);
+        if (str.len < idx || idx < 1) {
```

Comment:
Actually idx can be a negative number. eg. "123"[-2] which means the second last character in string "123". We should support this feature as duckdb also supports this.

Reply:
Done

Reply:
Added test for this as well